In [2]:
from pygbif import species, occurrences as occ
import pandas as pd

## Drive data

Let's take a look at the data uploaded to Google Drive at !2026-NABU-Asian-Hornet-Project > 01-Data

In [3]:
a_data = pd.read_csv('/Users/victoriacolombo/Documents/nabu/asian_hornet_DE.csv')
e_data= pd.read_csv('/Users/victoriacolombo/Documents/nabu/european_hornet_DE.csv')

/var/folders/xn/08v6g4r14lg5tpt_n37xl_5c0000gn/T/ipykernel_74511/1054268940.py:1: DtypeWarning: Columns (69,71,72,73,74,78,103) have mixed types. Specify dtype option on import or set low_memory=False.
  a_data = pd.read_csv('/Users/victoriacolombo/Documents/nabu/asian_hornet_DE.csv')
/var/folders/xn/08v6g4r14lg5tpt_n37xl_5c0000gn/T/ipykernel_74511/1054268940.py:2: DtypeWarning: Columns (75,81,82,95,101,103,104,105,109,110,111,113,114,115,116,117,118,119,120,121,122,123,126,127,128,129,130,132,133,134,135,136,137,138,139,140) have mixed types. Specify dtype option on import or set low_memory=False.
  e_data= pd.read_csv('/Users/victoriacolombo/Documents/nabu/european_hornet_DE.csv')


### 1. Size and date range

In [45]:
print('Asian hornet data shape:', a_data.shape, '\nEuropean hornet data shape:', e_data.shape, '\n\nAsian hornets date range:', a_data.eventDate.min(), 'to', a_data.eventDate.max(), '\nEuropean hornets date range:', e_data.eventDate.min(), 'to', e_data.eventDate.max())


Asian hornet data shape: (17866, 117) 
European hornet data shape: (61236, 141) 

Asian hornets date range: 2014-09-08 to 2025-12-30 
European hornets date range: 2000-05-05T00:00 to 2025-12-29T00:00


date is in yyyy-mm-dd format

there's date identified and event date

For european dataset event date 

European dataset has more columns than asian dataset (141 vs. 117)

In [20]:
a_data.identificationVerificationStatus.value_counts()

identificationVerificationStatus
Approved | Expert | Evidence    1064
Valid                            244
Approved | Expert                128
Approved | Automated              24
Non réalisable                     4
Name: count, dtype: int64

In [34]:
a_data.eventDate[0]

'2014-09-08'

In [33]:
a_data.dateIdentified.value_counts()

dateIdentified
2022-10-21T07:08:15    12
2024-11-25T10:29:49    11
2025-08-24T03:42:04     9
2024-10-18T07:17:14     9
2025-07-23T05:50:03     9
                       ..
2024-08-02T15:16:08     1
2024-08-03T14:07:10     1
2025-05-16T06:59:23     1
2024-08-02T15:15:49     1
2026-02-10T08:26:59     1
Name: count, Length: 4652, dtype: int64

In [26]:
a_data.month[0]

np.int64(9)

In [37]:
a_data.datasetName.value_counts()

datasetName
iNaturalist research-grade observations    711
Name: count, dtype: int64

countryCode corresponds to country when querying with GBIF
country is the full name of the country 
publishing country we don't need
species, species key
occurrence status?
scientific name = Vespa velutina Lepeletier, 1836 and Vespa velutina nigrithorax Buysson, 1905 assume they're the same?

identificationVerificationStatus. are all approved the same? is valid less than approved?
Approved | Expert | Evidence    1064
Valid                            244
Approved | Expert                128
Approved | Automated              24
Non réalisable                     4

gadm? 
{'level0': {'gid': 'DEU', 'name': 'Germany'}, 'level1': {'gid': 'DEU.11_1', 'name': 'Rheinland-Pfalz'}, 'level2': {'gid': 'DEU.11.18_1', 'name': 'Ludwigshafen am Rhein'}, 'level3': {'gid': 'DEU.11.18.1_1', 'name': 'Ludwigshafen am Rhein'}}


coordinateUncertaintyInMeters is this the error margin for lat and long?

date is in yyyy-mm-dd format

there's date identified and event date

For european dataset event date 

European dataset has more columns than asian dataset (141 vs. 117)

In [10]:
a_data.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17866 entries, 0 to 17865
Data columns (total 117 columns):
 #    Column                                 Dtype  
---   ------                                 -----  
 0    key                                    int64  
 1    datasetKey                             object 
 2    publishingOrgKey                       object 
 3    installationKey                        object 
 4    hostingOrganizationKey                 object 
 5    publishingCountry                      object 
 6    protocol                               object 
 7    lastCrawled                            object 
 8    lastParsed                             object 
 9    crawlId                                int64  
 10   extensions                             object 
 11   basisOfRecord                          object 
 12   occurrenceStatus                       object 
 13   classifications                        object 
 14   taxonKey                            

In [52]:
e_data.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 61236 entries, 0 to 61235
Data columns (total 141 columns):
 #    Column                                 Dtype  
---   ------                                 -----  
 0    key                                    int64  
 1    datasetKey                             object 
 2    publishingOrgKey                       object 
 3    installationKey                        object 
 4    hostingOrganizationKey                 object 
 5    publishingCountry                      object 
 6    protocol                               object 
 7    lastCrawled                            object 
 8    lastParsed                             object 
 9    crawlId                                int64  
 10   extensions                             object 
 11   basisOfRecord                          object 
 12   occurrenceStatus                       object 
 13   classifications                        object 
 14   taxonKey                            